File: Trees.cpp
#include<bits/stdc++.h>
using namespace std;

class node
{
	public:
		int data;
		node * left;
		node * right;
} * root = NULL;

void insert(node **root, int data)
{
	node *temp , *newnode;
	temp = *root;
	
	newnode = new node();
	newnode -> data = data;
	newnode -> left = NULL;
	newnode -> right = NULL;
	
	if(*root == NULL)
		*root = newnode;
	else
	{
		while(1)
		{
			if(data < temp -> data)
			{
				if(temp -> left != NULL)
				{
					temp = temp -> left;
				}
				else
				{
					temp -> left = newnode;
					break;
				}
			}
			else if(temp -> right != NULL)
			{
				temp = temp -> right;
			}
			else
			{
				temp -> right = newnode;
				break;
			}
		}
	}
}

void inorder(node *root)
{
	node *temp = root;
	if(temp == NULL)
		return;
	inorder(temp -> left);
	cout << temp -> data << " ";
	inorder(temp -> right);
}

void preorder(node *root)
{
	node *temp = root;
	if(temp == NULL)
		return;
		
	cout << temp -> data << " ";
	preorder(temp -> left);
	preorder(temp -> right);
}

void postorder(node *root)
{
	node *temp = root;
	if(temp == NULL)
		return;
	
	postorder(temp -> left);
	postorder(temp -> right);	
	cout << temp -> data << " ";

}



int main()
{
	int data;
	
	do
	{
		cin >> data;
		
		if(data > 0)
			insert(&root, data);
			
	} while(data > 0);
	
	cout << "\nInorder Traversal:\n";
	inorder(root);
	
	cout << "\nPreorder Traversal:\n";
	preorder(root);
	
	cout << "\nPostorder Traversal:\n";
	postorder(root);
	
}

File: Trees.java
import java.util.*;

class Trees
{
	static node temp;
	static node newnode;
	static class node
	{
		int data;
		node left;
		node right;
	} node root = null;

	public static Trees insert(Trees mroot, int data)
	{
		temp = mroot.root;
		
		newnode = new node();
		newnode.data = data;
		newnode.left = null;
		newnode.right = null;
		
		if(mroot.root == null)
			mroot.root = newnode;
		else
		{
			while(true)
			{
				if(data < temp.data)
				{
					if(temp.left != null)
					{
						temp = temp.left;
					}
					else
					{
						temp.left = newnode;
						break;
					}
				}
				else if(temp.right != null)
				{
					temp = temp.right;
				}
				else
				{
					temp.right = newnode;
					break;
				}
			}
		}
		return mroot;
	}

	static void inorder(node root)
	{
		node temp = root;
		if(temp == null)
			return;
		inorder(temp.left);
		System.out.print(temp.data +" ");
		inorder(temp.right);
	}

	static void preorder(node root)
	{
		node temp = root;
		if(temp == null)
			return;
		
		System.out.print(temp.data +" ");
		preorder(temp.left);
		preorder(temp.right);
	}

	static void postorder(node root)
	{
		node temp = root;  
		if(temp == null)
			return;
		
		postorder(temp.left);
		postorder(temp.right);	
		System.out.print(temp.data+" ");

	}



	public static void main(String args[])
	{
		int data;
		Trees mroot = new Trees();
		Scanner scan = new Scanner(System.in);
		do
		{
			data = scan.nextInt();
			if(data > 0)
				mroot = insert(mroot, data);
		} while(data > 0);
		
		System.out.print("\nInorder Traversal:\n");
		inorder(mroot.root);
		
		System.out.print("\nPreorder Traversal:\n");
		preorder(mroot.root);
		
		System.out.print("\nPostorder Traversal:\n");
		postorder(mroot.root);
		
	}

}








